lib/kernel/src/code_server.erl

```diff
@@ -1241,23 +1209,23 @@ loader_down(#state{loading = Loading0} = St, {Mod, Bin, FName}) ->
             St
     end.
 
+mod_to_bin([{Dir, nocache}|Tail], ModFile, Acc) ->
```

Comment:
Looking at this code, I think we should split the cache list from the path list.
In this function, every time we lookup a module we update the entire codepath - this might be quite a lot of work and extra allocations.
I think it would be quite beneficial for performance to split out the `Cache` as a map from the code path - this way once the cache is warmed up, this lookups will do no to little allocations.
This can be done as a follow-up PR, though, since this design of updating the codepath data-structure on every module lookup is present today already.

Reply:
Interesting!

>  I think it would be quite beneficial for performance to split out the Cache as a map from the code path - this way once the cache is warmed up, this lookups will do no to little allocations.

Would you have the map keys be the code paths? Then lookup is a bit more expensive. But I guess we could also build unique integers to associate them?

Reply:
> Another, as you suggested could be some manual interning and using the integer position in the code path list as the key to the map.

I was thinking I could simply have an always incrementing integer, this way there is no need to rehash as code paths are added or removed. :)

Reply:
That makes sense as well